### PR TITLE
Sync JVM flags with Trino defaults

### PR DESCRIPTION
**Pull request by nineinchnick**

### CODE REVIEW

**Comment by nineinchnick** [comment]
As expected, older versions fail to run with:
```
Unrecognized VM option 'G1NumCollectionsKeepPinned=10000000'
```
I could add a condition on the `image.tag`, but when using custom-built images, it doesn't have to be a version number. It's not possible to specify a different app version when installing the chart, so we can't use this field.

Maybe we should consider removing the default JVM flags from the chart and rely on those in the image, unless explicitly set?

**Review by mosabua — CHANGES_REQUESTED** [review]
I agree with the idea of removing the default config and relying on the docker container defaults. Users can still override it and we then have to just ensure its documented that you take the default from the container you use and update as desired.

File: charts/trino/templates/configmap-coordinator.yaml [hunk 1]
```diff
@@ -32,6 +32,14 @@ data:
     -Djdk.nio.maxCachedBufferSize=2000000
     # Allow loading dynamic agent used by JOL
     -XX:+EnableDynamicAgentLoading
+  {{/* only check the version number if the image is not overriden in any way */}}
+  {{- with .Values.image -}}
+  {{- if and (eq .repository "trinodb/trino") (not .useRepositoryAsSoleImageReference) (not .registry) (not .digest) (gt (default .Chart.AppVersion .tag | int ) 447 ) }}
```
Comment (nineinchnick) [inline charts/trino/templates/configmap-coordinator.yaml:37]: I've tested this on [the Helm playground](https://helm-playground.com/#t=LQhQCsDcFsDoGMD2A7AZgSwOYC4AEAfUXXAbxOFwHd0AXAC11gDUBDAGwFcBTAZ1nWgtMXXMAC%2BYoqXK50qXC2QATXAAouAR0YAnLgAdEPWom0BPXACIa29MkRKARgHprtxBYCUauzUYceXABK%2BobGZgCCPADKiGxcAJKCwsGoXLrI8Fxeqj46XJjoPNam2bmwSli8NNmYvqpKXKgsHGy%2BACxtAJyMNEIEssi%2BXh0A7LheElLEAMS4dDQ0ejzYTk4OHJh8iHpcyOBKANawJphr2oiUAU4AUgAiANLAABwAzABMnQCsb09TogAa-2wAGoAKrINiIeAHW7oIR2IroeBMACyAHk9DR0CgeH9gIDsABxACMADkONAAMKxOLwLE4%2B5cfQABVsyC4SgAvMSAAx8-k8qRkCi7FQSMhOABUsnkkqck2Iwtwotw4pIUqotAYcsmQA&v=LQhQEsFsEMHMFMBcoAEKDEKQoJIwSgE7yzgDOALoQJ4A0KAJvAGbQCuANhWShQPYp4kAA4U6KAO4ALcAGMpReGU7cU4AHYoAIn1kBreIQASbAEYo2ZOPFSLSlGohQAiZ7czYASvGF8y4fhoUDl1oCnA%2BTT5mXil4FAAVQg0BKGt6MWE5aA4OahQAAz5CWGh1cAAvMIj1AHo0hHVoSHgC22Jff0DqJyoUhlNavvU%2BdyxgXHx4ijh6JlYVHn5Y%2BKSUxQ54aDJ4gDdDf0iUMmF4WXBmcHgGFG3C6GFhADUDmoKUZkI%2BSBQAYSloIQKAA6ajNDi2GawJyuMbYADyohqOUY4AQlBQuxybHi0RWaimx1O50u11uPAKZABACYAKwANkQ0FMsgYwPZBWBKAAgkSzhcrjcsRwcSg%2BPtCMkmBSoZzbAw0UoKDC3GgPBMAOpxTRUHH0SJ5fGySIUeDqChqTQFDp%2BALFajvcgWHY3O4NeLEZiGM2yGxoSzwbydO00blkADKfE2eGs3i9xHUvqcrA4O1swk4HAAClG5D1cMwAHJ8ChZ4g7c2gIA) to make sure it works with arbitrary values in all the referenced fields.

**Review by wendigo — APPROVED** [review]

**Comment by nineinchnick** [comment]
I tried preparing another PR that would not override the default `jvm.config` but it's tricky, because we include values from the `.Values.coordinator.jvm` section. Handling this would require complex logic and IMO it's not worth it.

**Pull request merged**